### PR TITLE
Create profile

### PR DESCRIPTION
**Pull request by bravomike87**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,20 @@
 class User < ApplicationRecord
-  # has_one :profile
+  has_one :profile
   # has_many :artworks
```
Comment (AgP42) [inline app/models/user.rb]: why did you comment the "has_many" lines ?

File: app/views/profiles/edit.html.erb [hunk 2]
```diff
@@ -1,2 +1,11 @@
-<h1>Profiles#edit</h1>
-<p>Find me in app/views/profiles/edit.html.erb</p>
+<h1>Hello <%= @profile.first_name %></h1>
+<%= simple_form_for(@profile) do |f| %>
+    <%= f.input :first_name %>
+    <%= f.input :last_name %>
+    <%= "Email: (not changeable) #{@profile.user.email}" %>
+    <%= f.input :address %>
```
Comment (AgP42) [inline app/views/profiles/edit.html.erb:6]: do you want to change the databse schema to be able to store the address with the city on a separate field ?

File: app/views/profiles/edit.html.erb [hunk 3]
```diff
@@ -1,2 +1,11 @@
...
+    <%# <%= f.input :avatar %>
```
Comment (AgP42) [inline app/views/profiles/edit.html.erb]: for info : I updated the thello to add that we will need to update this profile form once we have the image uploader working

File: app/views/profiles/edit.html.erb [hunk 4]
```diff
@@ -1,2 +1,11 @@
...
+    <%#= f.input :avatar_cache. as :hidden %>
+    <%= f.submit "Save", class: "btn btn-primary" %>
```
Comment (AgP42) [inline app/views/profiles/edit.html.erb:10]: On the DB, we also have a description field on the profil page, with "Text" as a field type

File: app/controllers/profiles_controller.rb [hunk 5]
```diff
@@ -10,11 +11,28 @@ def create
...
+      redirect_to profile_path(@profile)
+    else
+      render :edit
+    end
   end
 
   def destroy
+    @profile = Profile.find(params[:id])
+    @profile.destroy
+    redirect_to Something_path
```
Comment (AgP42) [inline app/controllers/profiles_controller.rb]: maybe redirect on the root ?

File: app/controllers/profiles_controller.rb [hunk 6]
```diff
@@ -10,11 +11,28 @@ def create
...
   end
+
+  private
+
+  def profile_params
+    params.require(:profile).permit(:first_name, :last_name, :address, :avatar)
```
Comment (AgP42) [inline app/controllers/profiles_controller.rb]: and :description
(not yet on your form, see my other comment after)

File: app/models/profile.rb [hunk 7]
```diff
@@ -1,3 +1,6 @@
 class Profile < ApplicationRecord
   belongs_to :user
+  validates :first_name, presence: true, allow_nil: true
+  validates :last_name, presence: true, allow_nil: true
+  validates :address, presence: true, allow_nil: true
```
Comment (AgP42) [inline app/models/profile.rb:5]: great ! We add a task on thello with Peter this morning not to forget the validation, but much better to do it during the form creation as you did here !

**Review by AgP42 — COMMENTED** [review]
Great, thanks ! A few comments...

File: app/controllers/profiles_controller.rb (hunk 5 above)
Comment (googleheim) [reply app/controllers/profiles_controller.rb]: ok

File: app/views/profiles/edit.html.erb (hunk 2 above)
Comment (googleheim) [reply app/views/profiles/edit.html.erb:6]: let's keep the address as we have it for now

**Review by AgP42 — APPROVED** [review]

**Pull request merged**